front_end/audits2/Audits2Panel.js

```diff
@@ -1,43 +1,242 @@
+    [pwaLabel, bpLabel, perfLabel].forEach(label => {
+        label.checkboxElement.disabled = true;
+        auditSelectorForm.appendChild(label);
+    });
+
+    this._startButton = createTextButton(Common.UIString('Audit this page'), this._startButtonClicked.bind(this), 'run-audit audit-btn');
+    auditSelectorForm.appendChild(this._startButton);
+
+    this._statusView = this._createStatusView();
+    uiElement.appendChild(this._statusView);
+    return uiElement;
+  }
+
+  /**
+   * @return {!Element}
+   */
+  _createStatusView() {
+    var statusView = createElementWithClass('div', 'status hbox hidden');
+    statusView.createChild('span', 'icon');
+    this._statusElement = createElement('p');
+    statusView.appendChild(this._statusElement);
+    this._updateStatus('Loading...');
+    return statusView;
   }
 
   _start() {
-    SDK.targetManager.interceptMainConnection(this._dispatchProtocolMessage.bind(this)).then(rawConnection => {
-      this._rawConnection = rawConnection;
-      this._send('start').then(result => {
-        var section = new Components.ObjectPropertiesSection(
-            SDK.RemoteObject.fromLocalObject(result), Common.UIString('Audit Results'));
-        this._resultElement.appendChild(section.element);
-        this._stop();
-      });
+    this._auditRunning = true;
+    this._updateButton();
+    this._updateStatus('Loading...');
+
+    this._inspectedURL = SDK.targetManager.mainTarget().inspectedURL();
+
+    return Promise.resolve()
+        .then(_ => this._protocolService.attach())
+        .then(_ => this._protocolService.startLighthouse(this._inspectedURL))
+        .then(this._finish.bind(this));
+  }
+
+  /**
+   * @param {!Event} event
+   */
+  _startButtonClicked(event) {
+    if (this._auditRunning) {
+      this._updateStatus('Cancelling...');
+      this._stop();
+      return;
+    }
+    this._start();
+  }
+
+  _updateButton() {
+    this._startButton.textContent = this._auditRunning ? Common.UIString('Cancel audit') : Common.UIString('Audit this page');
+    this._startButton.classList.toggle('started', this._auditRunning);
+    this._statusView.classList.toggle('hidden', !this._auditRunning);
+  }
+
+  /**
+   * @param {string} statusMessage
+   */
+  _updateStatus(statusMessage) {
+    this._statusElement.textContent = Common.UIString(statusMessage);
+  }
+
+  _stop() {
+    this._protocolService.detach().then(_ => {
+      this._auditRunning = false;
+      this._updateButton();
+      if (this._inspectedURL !== SDK.targetManager.mainTarget().inspectedURL())
+          SDK.targetManager.mainTarget().pageAgent().navigate(this._inspectedURL);
+
     });
   }
 
   /**
+   * @param {!Audits2.WorkerResult} result
+   */
+  _finish(result) {
+    this._stop();
+    this.resultsView.removeChildren();
+
+    this.resultsView.appendChild(this._createResultsBar(result.lighthouseResult.url, result.lighthouseResult.generatedTime));
```

Comment:
> `WARNING - actual parameter 1 of Audits2.Audits2Panel.prototype._createResultsBar does not match formal parameter`

this is confusing. I don't see how `url` could be `(null|string)`, *but* you are doing a property access on something (`result.lighthouseResult`) you declare above could be null. If you change that in the `Audits2.WorkerResult` typedef that might help here